google-api-client/src/main/java/com/google/api/client/googleapis/services/AbstractGoogleClientRequest.java

```diff
@@ -119,6 +124,48 @@ protected AbstractGoogleClientRequest(AbstractGoogleClient abstractGoogleClient,
     } else {
       requestHeaders.setUserAgent(USER_AGENT_SUFFIX);
     }
+    // Set the header for the Api Client version (Java and OS version)
+    requestHeaders.set(API_VERSION_HEADER, ApiClientVersion.build(abstractGoogleClient));
+  }
+
+  /**
+   * Internal class to help build the X-Goog-Api-Client header. This header identifies the
+   * API Client version and environment.
+   *
+   * See <a href="https://cloud.google.com/apis/docs/system-parameters"></a>
+   *
+   */
+  private static class ApiClientVersion {
+    private static final String JAVA_VERSION = formatSemver(System.getProperty("java.version"));
+    private static final String OS_NAME = formatName(System.getProperty("os.name"));
+    private static final String OS_VERSION = formatSemver(System.getProperty("os.version"));
+
+    private static String build(AbstractGoogleClient client) {
+      // TODO(chingor): add the API version from the generated client
```

Comment:
I _think_ this is somehow already happening, or partially happening. Happy to discuss offline if you care to investigate.